engine/src/flutter/lib/web_ui/lib/painting.dart

```diff
@@ -613,16 +613,24 @@ abstract class FrameInfo {
   Image get image;
 }
 
+typedef CodecEventCallback = void Function(Codec codec);
+
 class Codec {
-  Codec._();
+  Codec._() {
+    onCreate?.call(this);
+  }
+  static CodecEventCallback? onCreate;
+  static CodecEventCallback? onDispose;
   int get frameCount => 0;
   int get repetitionCount => 0;
   Future<FrameInfo> getNextFrame() {
     return engine.futurize<FrameInfo>(_getNextFrame);
   }
 
   String? _getNextFrame(engine.Callback<FrameInfo> callback) => null;
-  void dispose() {}
+  void dispose() {
+    onDispose?.call(this);
+  }
```

Comment:
It's probably a bug/mistake that the class is not marked as `interface`. All instance methods should be abstract. The concrete implementations of the `dispose` method are in subclasses. `onDispose` should be called in each implementation individually (unless we make it a `base` class or something and force subclasses to call `super.dispose()`):

![Screenshot 2025-01-23 at 3 06 39 PM](https://github.com/user-attachments/assets/12446bcf-6e94-4d74-8e32-45bea7125261)